pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -107,6 +115,8 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 		ReportDeviceInfo.Ncpu = *proto.Uint32(metric.Ncpus)
 		ReportDeviceInfo.Memory = *proto.Uint64(metric.TotalMemoryMB)
 	}
+
+	ReportDeviceInfo.PowerCycleCounter = getSMARTAttributeValue(SMARTPowerCycleCount)
```

Comment:
We should only need to retrieve this on boot, since it doesn't change past that.

I wonder if this can take a while thus whether we should run it separate goroutine kicked off at startup and which exits when done.

Don't know if we'll add other smart counters (to get time series on disk/ssd health) and if so whether we'd want to do all of it in volumemgr. But perhaps we'll introduce some hardwaremgr to do that (and also retrieve temperature sensors etc in such an agent). So doing it at startup in zedagent is fine for now.

Reply:
Fetching PowerCycleCounter as part of a goroutine which is triggered at the start of zedagent.

Reply:
OK